applications/crossbar/src/modules/cb_notifications.erl

```diff
@@ -298,6 +298,23 @@ do_post(Context) ->
         _Status -> Context1
     end.
 
+-spec set_system_macros(cb_context:context()) -> cb_context:context().
+-spec set_system_macros(cb_context:context(), api_binary()) -> cb_context:context().
+set_system_macros(Context) ->
+    Id = wh_doc:id(cb_context:doc(Context)),
+    set_system_macros(Context, Id).
+
+set_system_macros(Context, 'undefined') -> Context;
+set_system_macros(Context, Id) ->
+    SysContext = read_system(Context, Id),
+    SysDoc = cb_context:doc(SysContext),
+    Macros = wh_json:get_value(?MACROS, SysDoc, wh_json:new()),
+    Updater =
+        fun(JObj) ->
+            wh_json:set_value(?MACROS, Macros, JObj)
+        end,
+    cb_context:update_doc(Context, Updater).
```

Comment:
Why not just call `cb_context:set_doc(Context, wh_json:set_value(?MACROS, Macros, JObj))` directly?